File: jose/3.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main ()
{
    int x = 0;
    printf("Digite um numero HEX: ");
    scanf("%x", &x);
    
    
    char str[10] = {0};
    int size = snprintf(str, sizeof(str), "%d", x);
    printf("Dec: %s, size: %d\n", str, size);


    int espelho = 1;
    for (int i = 0; i < size; i++)
    {
        if (str[0+i] != str[size-(i+1)])
            {espelho = 0; break;}
    }
    
    
    if (espelho) {printf("S");}
    else {printf("N");}
    

    return 0;
}

File: jose/python/3.py
n = int(input("Digite o numero: "), 16)

txt = str(n)
txt_inv = ""

for i in range(len(txt)-1, -1, -1):
    txt_inv += txt[i]

print(txt, txt_inv)

if (txt == txt_inv):
    print("S")
else:
    print("N")
